src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1215,6 +1221,30 @@ private TimeZoneInfo(byte[] data, string id, Boolean dstDisabled)
 
             ValidateTimeZoneInfo(m_id, m_baseUtcOffset, m_adjustmentRules, out m_supportsDaylightSavingTime);
         }
+
+        private void GetDisplayName(Interop.GlobalizationInterop.TimeZoneDisplayNameType nameType, ref string displayName)
+        {
+            string timeZoneDisplayName;
+            bool result = Interop.CallStringMethod(
+                (locale, id, type, stringBuilder) => Interop.GlobalizationInterop.GetTimeZoneDisplayName(
+                    locale,
+                    id,
+                    type,
+                    stringBuilder,
+                    stringBuilder.Capacity),
+                CultureInfo.CurrentUICulture.Name,
+                m_id,
+                nameType,
+                out timeZoneDisplayName);
+
+            // If there is an unknown error, don't set the displayName field.
+            // It will be set to the abbreviation that was read out of the tzfile.
```

Comment:
We can't assert that displayName is not null because there are a few time zones that don't have any DaylightSavings transitions in them.  Thus they won't have a daylightName from the abbreviations in the tzfile.